docs/spec/spec-proposals/f1-fee-distribution/f1_fee_distr.tex

```diff
@@ -17,49 +17,69 @@
+The trivial solution of just giving the rewards to each delegator every block is too expensive to perform on-chain.
+So instead fee distribution algorithms have delegators perform an explicit withdraw action, which yields the same total amount of fees as though they received it at every block. 
+
+This details F1, an approximation-free, slash-tolerant fee distribution algorithm which allows validator commission-rates, inflation rates, and fee proportions, which can all efficiently change per validator, every block.
+The algorithm requires iterating over the validators every block, and withdraws require iterating over all of the corresponding validator's slashes whilst the delegator was bonded.
+The former iteration is cheap, due to staking logic already requiring iteration over all validators, which causes the expensive state-reads to be cached.
+The number of slashes is expected to be 0 or 1 for most validators,
+so the latter term also meets the blockchains efficiency needs.
+
+The key point of how F1 works is that it tracks how much rewards a delegator with 1 stake for a given validator would be entitled to if it had bonded at block 0 until the latest block.
+When a delegator bonds at block $b$, the amount of rewards a delegator with 1 stake would have if bonded at block 0 until block $b$ is also persisted to state.
+When the delegator withdraws, they receive the difference of these two values.
+Since rewards are distributed according to stake-weighting, this amount of rewards can be scaled by the amount of stake a delegator had.
+Section 1.2 describes this in more detail, with an argument for it being approximation free. 
+Section 2 details how to adapt this algorithm to handle commission rates, slashing, and inflation.
+
+\subsection{Base algorithm}
+In this section, we show that the F1 base algorithm gives each delegator rewards identical to that which they'd receive in the naive and correct fee distribution algorithm that iterated over all delegators every block.
+
+Even distribution of a validators rewards amongst its validators weighted by stake means the following: 
+Suppose a delegator delegates $x$ stake to a validator $v$ at block $h$.
 Let the amount of stake the validator has at block $i$ be $s_i$ and the amount of fees they receive at this height be $f_i$.
 Then if a delegator contributing $x$ stake decides to withdraw at block $n$, the rewards they receive is 
 $$\sum_{i = h}^{n} \frac{x}{s_i}f_i = x \sum_{i = h}^{n} \frac{f_i}{s_i}$$
 
-However $s_i$ will not change every block.
-It only changes if the validator gets slashed, or if someone new has bonded or unbonded.
-Handling slashes is relegated to \autoref{ssec:slashing}.
-Define a period as the set of blocks between two changes in a given validator's total stake.
+Note that $s_i$ does not change every block,
+it only changes if the validator gets slashed, or if someone new has bonded or unbonded.
```

Comment:
This seems wrong, `s_i` should change if any delegator to the validator changes the amount of stake they are delegating. Is that what you mean?

Reply:
Yeah. Is there a succinct way to describe this?

Reply:
"if the validator gets slashed, or if any delegator alters the amount of their bond" perhaps